### PR TITLE
[PM-4048] Update DeleteAccountComponent on Desktop

### PR DESCRIPTION
**Pull request by andrebispo5**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

[PM-4048](https://bitwarden.atlassian.net/browse/PM-4048)
Convert the usage of ModalService to DialogService.

## Code changes

**apps/desktop/src/app/app.component.ts**: Change modalService to component static method with `dialogService`.
**apps/desktop/src/app/app.module.ts**: Change import from angular to our core DialogModule, which also exports @angular/cdk/dialog. Updated standalone component imports.
**apps/desktop/src/auth/delete-account.component.html:** convert UI to use `bit-dialog` and other CL directives like `bitSubmit` and `bitFormButton`
**apps/desktop/src/auth/delete-account.component.ts:** Added static open per migration docs. Refactor submit(...) to arrow func to work with bitSubmit + added manual error handling as it was lost in the change. Converted to standalone component
**apps/desktop/src/app/components/user-verification.component.ts**: Converted to standalone component

## Screenshots

image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

[PM-4048]: https://bitwarden.atlassian.net/browse/PM-4048?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/components/src/dialog/dialog.module.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ import { IconDirective, SimpleDialogComponent } from "./simple-dialog/simple-dia
     SimpleDialogComponent,
     DialogCloseDirective,
     IconDirective,
+    ButtonModule,
```
Comment (Hinton) [inline libs/components/src/dialog/dialog.module.ts:40]: If you want to propose adding `ButtonModule` to the `DialogModule` please do so in a separate PR for proper review.

Comment (JaredSnider-Bitwarden) [reply libs/components/src/dialog/dialog.module.ts]: I've refactored the code to avoid this added import.

File: apps/desktop/src/auth/delete-account.component.html [hunk 2]
```diff
@@ -1,43 +1,35 @@
...
-          </div>
-          <div id="confirmIdentityHelp" class="box-footer">
-            <p>{{ "confirmIdentity" | i18n }}</p>
-          </div>
-        </div>
+<form #form [bitSubmit]="submit" [formGroup]="deleteForm">
+  <bit-dialog>
+    <div bitDialogTitle>{{ "deleteAccount" | i18n }}</div>
+    <div bitDialogContent class="box">
+      <p class="modal-text">{{ "deleteAccountDesc" | i18n }}</p>
```
Comment (willmartian) [inline apps/desktop/src/auth/delete-account.component.html:5]: Remove non Tailwind classes

File: apps/desktop/src/auth/delete-account.component.html [hunk 3]
```diff
@@ -1,43 +1,35 @@
...
+      <bit-callout type="warning" title="{{ 'warning' | i18n }}">
+        {{ "deleteAccountWarning" | i18n }}
+      </bit-callout>
+      <div class="box-content">
```
Comment (willmartian) [inline apps/desktop/src/auth/delete-account.component.html:9]: Remove non Tailwind classes

File: apps/desktop/src/auth/delete-account.component.html [hunk 4]
```diff
@@ -1,43 +1,35 @@
...
       </div>
-      <div class="modal-footer">
-        <button type="submit" class="danger" [disabled]="form.loading || !secret">
-          <i class="bwi bwi-spinner bwi-spin" [hidden]="!form.loading" aria-hidden="true"></i>
-          <span [hidden]="form.loading">{{ "deleteAccount" | i18n }}</span>
-        </button>
-        <button type="button" data-dismiss="modal" [disabled]="form.loading">
-          {{ "cancel" | i18n }}
-        </button>
+      <div id="confirmIdentityHelp" class="box-footer">
```
Comment (willmartian) [inline apps/desktop/src/auth/delete-account.component.html:16]: Remove non Tailwind class

File: apps/desktop/src/auth/delete-account.component.html [hunk 5]
```diff
@@ -1,43 +1,35 @@
...
+    </div>
+    <div bitDialogFooter>
+      <button bitButton bitFormButton type="submit" buttonType="danger" [disabled]="!secret">
+        {{ "deleteAccount" | i18n }}
+      </button>
+      <button
+        bitButton
+        type="button"
+        buttonType="secondary"
+        [disabled]="form.loading"
```
Comment (willmartian) [inline apps/desktop/src/auth/delete-account.component.html:28]: Replace this with `bitFormButton`, which handles the loading and disabled state.

File: apps/desktop/src/auth/delete-account.component.ts [hunk 6]
```diff
@@ -6,14 +7,15 @@ import { Verification } from "@bitwarden/common/auth/types/verification";
 import { I18nService } from "@bitwarden/common/platform/abstractions/i18n.service";
 import { LogService } from "@bitwarden/common/platform/abstractions/log.service";
 import { PlatformUtilsService } from "@bitwarden/common/platform/abstractions/platform-utils.service";
+import { ValidationService } from "@bitwarden/common/platform/abstractions/validation.service";
+
+import { DialogService } from "../../../../libs/components/src/dialog";
```
Comment (willmartian) [inline apps/desktop/src/auth/delete-account.component.ts:12]: Use import alias `from "@bitwarden/components"`

File: apps/desktop/src/auth/delete-account.component.html [hunk 7]
```diff
@@ -1,43 +1,35 @@
...
-              formControlName="verification"
-              name="verification"
-            >
-            </app-user-verification>
-          </div>
-          <div id="confirmIdentityHelp" class="box-footer">
-            <p>{{ "confirmIdentity" | i18n }}</p>
-          </div>
-        </div>
+<form #form [bitSubmit]="submit" [formGroup]="deleteForm">
```
Comment (willmartian) [inline apps/desktop/src/auth/delete-account.component.html:1]: Is `#form` still needed?

File: apps/desktop/src/auth/delete-account.component.ts [hunk 8]
```diff
@@ -23,25 +24,30 @@ export class DeleteAccountComponent {
...
         "success",
         this.i18nService.t("accountDeleted"),
         this.i18nService.t("accountDeletedDesc")
       );
     } catch (e) {
+      this.validationService.showError(e);
       this.logService.error(e);
     }
-  }
+  };
```
Comment (willmartian) [inline apps/desktop/src/auth/delete-account.component.ts:52]: Technically, we could remove the entire `try/catch` here. `bitSubmit` catches errors and handles them the same way: https://github.com/bitwarden/clients/blob/4b1570b0b302f15b685c861828c733f8ab6ab260/libs/components/src/async-actions/bit-submit.directive.ts#L46-L48

File: apps/desktop/src/auth/delete-account.component.html [hunk 9]
```diff
@@ -1,43 +1,29 @@
...
-            </app-user-verification>
-          </div>
-          <div id="confirmIdentityHelp" class="box-footer">
-            <p>{{ "confirmIdentity" | i18n }}</p>
-          </div>
-        </div>
+<form [bitSubmit]="submit" [formGroup]="deleteForm">
+  <bit-dialog>
+    <div bitDialogTitle>{{ "deleteAccount" | i18n }}</div>
+    <div bitDialogContent class="box">
```
Comment (willmartian) [inline apps/desktop/src/auth/delete-account.component.html:4]: Remove non-Tailwind class `box`

(sorry, missed this one on last review)

File: apps/desktop/src/auth/delete-account.component.ts (hunk 8 above)
Comment (JaredSnider-Bitwarden) [reply apps/desktop/src/auth/delete-account.component.ts:52]: Great catch! I tested to confirm this as well.

**Review by willmartian — APPROVED** [review]
MIgration LGTM. I also:
 
1. added a border to the `user-verification.component` to up the contrast--it looked a little wonky since it isn't migrated to the CL yet. [fed827a](https://github.com/bitwarden/clients/pull/6846/commits/fed827acf10193f4d29907a50c699a994ccb8c3f) (Screenshot updated to reflect this change)
2. Made both of the components standalone, to prevent adding CL imports to the desktop `AppModule` [9181ca4](https://github.com/bitwarden/clients/pull/6846/commits/9181ca4af6706aa6828bdeffcb69fbde41e75c0d)

**Review by JaredSnider-Bitwarden — APPROVED** [review]
This looks great! Thank you for your help @willmartian!

**Comment by JaredSnider-Bitwarden** [comment]
Passed QA, merging!
